qiskit/passmanager/base_tasks.py

```diff
@@ -103,7 +103,7 @@ def execute(
             run_state = RunState.FAIL
             raise
         finally:
-            ret = ret or passmanager_ir
+            ret = passmanager_ir if ret is None else ret
```

Comment:
Strictly this doesn't account for the possibility of `None` being a valid value within the backend's IR - we'd revert that as well if it were.  If you care about handling that case, we could make a sentinel `_UNSET = object()` global and then set `ret = _UNSET` and compare to `if ret is _UNSET` later.